pkg/manager/impl/execution_manager.go

```diff
@@ -332,11 +332,20 @@ func (m *ExecutionManager) setCompiledTaskDefaults(ctx context.Context, task *co
 		logger.Warningf(ctx, "Can't set default resources for nil task.")
 		return
 	}
-	if task.Template == nil || task.Template.GetContainer() == nil || task.Template.GetContainer().Resources == nil {
+	if task.Template == nil || task.Template.GetContainer() == nil {
 		// Nothing to do
 		logger.Debugf(ctx, "Not setting default resources for task [%+v], no container resources found to check", task)
 		return
 	}
+
+	if task.Template.GetContainer().Resources == nil {
+		// In case of no resources on the container, create empty requests and limits
+		// so the container will still have resources configure properly
+		task.Template.GetContainer().Resources = &core.Resources{
+			Requests: []*core.Resources_ResourceEntry{},
```

Comment:
Do we handle it correctly if the resources is not nil but requests and limits are?

Reply:
I haven't check but if the resource is not nil then requests and limits should be empty arrays, right? In that case, there is not problem as `assignResourcesIfUnset` just loop through then

Reply:
@EngHabu . I have a question about this [line](https://github.com/flyteorg/flyteadmin/blob/ab2087db162412e80a690f09c701aa6da8764d78/pkg/manager/impl/execution_manager.go#L365). Why it is not using the `limit` under `TaskResourceConfiguration` which I think should be the same for the [request](https://github.com/flyteorg/flyteadmin/blob/ab2087db162412e80a690f09c701aa6da8764d78/pkg/manager/impl/execution_manager.go#L357). This function [`createTaskDefaultLimits `](https://github.com/flyteorg/flyteadmin/blob/ab2087db162412e80a690f09c701aa6da8764d78/pkg/manager/impl/execution_manager.go#L212) reads the defaults of resource and take that as `platformValues`.

Reply:
hey @tnsetting i think the thinking was we want to use the request values by default if the limit values are unset - but you're right, at this point they'll always be set with either user-provided values or the platform request defaults. mind filing a bug so we can rework this? unfortunately i think this whole code block has grown quite convoluted

Reply:
@tnsetting can we add a test for haytham's concern?

Reply:
@katrogan I am working on testing the requests/limits for the compiled tasks. In case of flytekit-java tasks, since resources is `nil` which means both requests and limits are empty, the requests is set to the default from the config file. The problem is that the limits is also set to the default in the config file which is a bit problematic. I am not sure what limits value I should use since you don't want to use the limits from the config.

I can add two test cases where one of the `requests` and `limits` is missing.